overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,211 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	SnapName        string          `json:"snap-name"`
+	ExportedVersion string          `json:"exported-version"`
+	Symlinks        []SymlinkExport `json:"symlinks,omitempty"`
+}
+
+// NewManifestForSnap returns the manifest of a given snap.
+//
+// Currently only the core and snapd snaps export any content to the system. As
+// such the export manifest is not embedded into the snap meta-data but instead
+// computed here.
+//
+// Both snapd and core snaps export content under the snap name "snapd", using
+// the export set name "tools". The revision is mangled, for "snapd" it is used
+// directly. For "core" it is transformed to "core_$revision".
+func NewManifestForSnap(info *snap.Info) *Manifest {
+	// XXX: should we use WellKnownSnapID here? Probably not as this must work for
+	// unsigned snaps as well. Alternatively, should we look at snap type, as we
+	// have unique values for both snapd and core.
```

Comment:
yes, I think this sort of code nowadays should use types

Reply:
Switched to types